File: works/16917.java
import java.util.Scanner;

public class Main
{
    public static void main(String[] args)
    {
        Scanner sc = new Scanner(System.in);
        int A, B, C, X, Y;
        A = sc.nextInt(); B = sc.nextInt(); C = sc.nextInt();
        X = sc.nextInt(); Y = sc.nextInt();
        int cost = 0;
        
        while(true)
        {
            int x=1, y=1;
            if(X <= 0) x = 0;
            if(Y <= 0) y = 0;

            int a = cost + A*x + B*y;
            int b = cost + C * 2;
            cost = Math.min(a,b);

            X--; Y--;
            if(X < 0 && Y < 0) break;
        }
        System.out.print(cost);
    }
}

File: works/16917.kt
import java.util.*

fun main()
{
    val sc = Scanner(System.`in`)
    var A = sc.nextInt()
    var B = sc.nextInt()
    var C = sc.nextInt()
    var X = sc.nextInt()
    var Y = sc.nextInt()
    var cost = 0

    while(true)
    {
        var x = 1
        var y = 1
        if(X <= 0) x = 0
        if(Y <= 0) y = 0

        var a = cost + A*x + B*y
        var b = cost + C * 2
        cost = Math.min(a, b)

        X--; Y--
        if(X < 0 && Y < 0) break
    }
    print(cost)
}